sale_loyalty_order_line_link/models/sale_order.py

```diff
@@ -176,9 +151,9 @@ def _filter_related_program_lines(self, program):
         the product criteria rules, we can extend this method to return the proper
         records."""
         # No domain means that any product meets the promotion rules
-        if not program.rule_products_domain:
-            return self
-        domain = safe_eval(program.rule_products_domain)
+        if not program.program_id.rule_ids.product_domain:
```

Comment:
This got me thinking for a while :sweat_smile: as the nomenclature is really misleading. If you're going to use rewards use rewards, but don't name it otherwise.

And I'd say you shouldn't as you're doing an unnecessary roundtrip

Reply:
You're right it wasn't properly named or correct